# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../busybox/dietlibc.patch
# Copyright (C) 2007 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./shell/ash.c.orig	2007-07-17 11:01:59.000000000 -0400
+++ ./shell/ash.c	2007-07-17 11:02:57.000000000 -0400
@@ -66,6 +66,9 @@
 #error "Do not even bother, ash will not run on uClinux"
 #endif
 
+#ifdef __dietlibc__
+#define dprintf fdprintf
+#endif
 
 /* ============ Misc helpers */
 
--- ./libbb/pw_encrypt.c.orig	2007-07-21 19:35:23.000000000 -0400
+++ ./libbb/pw_encrypt.c	2007-07-21 19:35:36.000000000 -0400
@@ -8,7 +8,7 @@
  */
 
 #include "libbb.h"
-#include <crypt.h>
+#include <unistd.h>
 
 char *pw_encrypt(const char *clear, const char *salt)
 {
